### PR TITLE
689 - Update output to print to sysout while refusing all errors or exceptions that are not validation related from going to the front end

### PR DESCRIPTION
**Pull request by sasha-dresden**

### CODE REVIEW

File: django-backend/fecfiler/utils.py [hunk 1]
```diff
@@ -19,8 +19,11 @@ def custom_exception_handler(exc, context):
     if response is not None and response.status_code == 403:
         delete_user_logged_in_cookies(response)
 
-    # Do not allow an error response body
-    # if getattr(response, 'data'):
-    #     response.data = None
+    # Do not allow an error response body unless validation
+    data = getattr(response, 'data')
+    exception_type = type(exc)
+    print(data)
```
Comment (lbeaufort) [inline django-backend/fecfiler/utils.py:25]: @sasha-dresden I think we should avoid using `print` statements

**Review by lbeaufort — CHANGES_REQUESTED** [review]
I think we should avoid using `print` statements

File: django-backend/fecfiler/utils.py (hunk 1 above)
Comment (sasha-dresden) [reply django-backend/fecfiler/utils.py]: Good point. Switched it over to use structlog

File: django-backend/fecfiler/utils.py [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
         delete_user_logged_in_cookies(response)
 
-    # Do not allow an error response body
-    # if getattr(response, 'data'):
-    #     response.data = None
+    # Do not allow an error response body unless validation
+    data = getattr(response, 'data')
+    exception_type = type(exc)
+    logger.error("Error: {}".format(data))
+    if data and "ValidationError" not in str(exception_type):
```
Comment (toddlees) [inline django-backend/fecfiler/utils.py:29]: Can avoid string parsing by using `exception_type is not ValidationError`

importing:
`from rest_framework.exceptions import ValidationError`

File: django-backend/fecfiler/utils.py [hunk 3]
```diff
@@ -1,26 +1,32 @@
...
+    if response.status_code == 403:
         delete_user_logged_in_cookies(response)
 
-    # Do not allow an error response body
-    # if getattr(response, 'data'):
-    #     response.data = None
+    # Do not allow an error response body unless validation
+    data = getattr(response, 'data')
+    exception_type = type(exc)
+    logger.error("Error: {}".format(data))
```
Comment (toddlees) [inline django-backend/fecfiler/utils.py:28]: 3.6 introduced f-strings which tend to be more readable https://peps.python.org/pep-0498/
`logger.error(f"Error: {data}")`

**Review by toddlees — CHANGES_REQUESTED** [review]

File: django-backend/fecfiler/utils.py (hunk 3 above)
Comment (sasha-dresden) [reply django-backend/fecfiler/utils.py]: Ah that was a simple enough update and a lot cleaner

File: django-backend/fecfiler/utils.py (hunk 2 above)
Comment (sasha-dresden) [reply django-backend/fecfiler/utils.py]: Oh I think I tried that previously but I imported the wrong ValidationError, I think I tried importing from django.forms and that didn't seem to work so I just used string comparison. But class validation is much better. Got that updated with the correct ValidationError now.

**Review by toddlees — APPROVED** [review]

**Pull request merged**